examples/rivets/js/stores/todo-store.js

```diff
@@ -0,0 +1,89 @@
+(function (exports) {
+	'use strict';
+
+	exports.TodoStore = {
+		// All todos.
+		all: JSON.parse(localStorage['todos-rivets'] || '[]'),
+
+		// Currently applied filter.
+		statusFilter: '',
+
+		// Returns the filtered todo list.
+		filtered: function () {
+			var status = this.statusFilter;
+
+			return this.all.filter(function (todo) {
+				switch (status) {
+					case 'active':
+						return !todo.completed;
+					case 'completed':
+						return todo.completed;
+					default:
+						return true;
+				}
+			});
+		},
+
+		// Returns all active todos.
+		active: function () {
+			return this.all.filter(function (todo) {
+				return !todo.completed;
+			});
+		},
+
+		completed: function () {
+			return this.all.filter(function (todo) {
+				return todo.completed;
+			});
+		},
+
+		// Adds a new todo by title and saves.
+		add: function (todo) {
+			this.all.push(todo);
+			this.save();
+		},
+
+		// Removes a todo and saves.
+		remove: function (todo) {
+			this.all.splice(this.all.indexOf(todo), 1);
+			this.save();
+		},
+
+		// Toggles the todo as complete / active and saves.
+		toggle: function (todo) {
+			todo.completed = !todo.completed;
+			this.save(true);
```

Comment:
as @sindresorhus pointed out last week to me

http://ariya.ofilabs.com/2011/08/hall-of-api-shame-boolean-trap.html